modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/TWKBAttributeIO.java

```diff
@@ -0,0 +1,231 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2019, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ */
+
+package org.geotools.data.postgis;
+
+import java.io.IOException;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import org.geotools.data.Base64;
+import org.geotools.data.DataSourceException;
+import org.geotools.geometry.jts.LiteCoordinateSequence;
+import org.geotools.geometry.jts.LiteCoordinateSequenceFactory;
+import org.locationtech.jts.geom.CoordinateSequence;
+import org.locationtech.jts.geom.CoordinateSequences;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.LineString;
+import org.locationtech.jts.geom.LinearRing;
+import org.locationtech.jts.geom.MultiLineString;
+import org.locationtech.jts.geom.MultiPolygon;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.geom.Polygon;
+import org.locationtech.jts.io.ByteArrayInStream;
+import org.locationtech.jts.io.WKBWriter;
+
+public class TWKBAttributeIO {
+    TWKBReader wkbr;
+    ByteArrayInStream inStream = new ByteArrayInStream(new byte[0]);
+    GeometryFactory gf;
+
+    /**
+     * A variation of TWKBReader tha optimizes conversion of collapsed point expressed using
+     * LiteCoordinateSequece
```

Comment:
Typo:)  Should be "LiteCoordinateSequence".  (Had to find something to prove that I read the PR.;))

Reply:
:-D